lib/tasks/scripts/clean_biswanath_dupes.rb

```diff
@@ -0,0 +1,248 @@
+  #   * BusinessIdentifier (confirm with mobile team if phone number edit is in place)
+  #   * PatientPhoneNumber (confirm with mobile team if phone number edit is in place)
+
+  def self.call(*args)
+    new(*args).call
+  end
+
+  attr_accessor :dryrun, :exact_matches, :ambiguous_matches, :no_matches
+
+  def initialize(dryrun: false)
+    @dryrun = dryrun
+    @exact_matches = {}
+    @ambiguous_matches = []
+    @no_matches = []
+  end
+
+  def call
+    puts 'Identifying patient matches...'
+    identify_patient_matches
+
+    return if dryrun
+
+    puts 'Porting activity of patients with exact matches...'
+    port_exact_match_activity
+
+    puts 'Porting unmatched patients to the import user...'
+    port_unmatched_patients
+
+    puts 'Deactiviting ambiguous patients...'
+    deactivate_ambiguous_patients
+
+    puts 'Complete. Goodbye.'
+  end
+
+  def identify_patient_matches
+    duplicate_patients.each do |patient|
+      matches = match_by_name_age(patient)
+      if matches.count == 0
+        (no_matches << patient) && next
+      end
+
+      if matches.count == 1
+        (exact_matches[patient.id] = matches.first.id) && next
+      end
+
+      matches = match_by_name_age_address(patient)
+      if matches.count == 1
+        (exact_matches[patient.id] = matches.first.id) && next
+      end
+
+      matches = match_by_name_age_address_phone(patient)
+      if matches.count == 1
+        (exact_matches[patient.id] = matches.first.id) && next
+      end
+
+      ambiguous_matches << patient
+    end
+
+    print_summary
+  end
+
+  def port_exact_match_activity
+    duplicate_patients = Patient.where(id: exact_matches.keys)
+
+    no_action_required = duplicate_patients.select do |patient|
+      no_activity?(patient) && no_modifications?(patient)
+    end
+    no_action_required.each(&:discard_data)
+    puts "- #{no_action_required.count} patients have no activity or modifications"
+
+    actionable_patients = duplicate_patients - no_action_required
+
+    puts "- #{actionable_patients.count} patients have new activity or modifications"
+
+    bp_denylist = [
+      '3162e5db-e959-4c3e-af64-5d703ba1dc5a',
+      '081530b8-60d9-47c6-bddf-ac0fe6efc7d4',
+      'e409c013-b52d-4a87-98cf-5088fb31b705'
+    ]
+
+    actionable_patients.each do |patient|
+      real_patient_id = exact_matches[patient.id]
+
+      blood_pressures = patient.blood_pressures.where.not(id: bp_denylist).each { |record| record.update!(patient_id: real_patient_id) }
+      blood_sugars = patient.blood_sugars.each { |record| record.update!(patient_id: real_patient_id) }
+
+      port_bp_encounters(real_patient_id, blood_pressures)
+      port_bs_encounters(real_patient_id, blood_sugars)
+      patient.appointments.each { |record| record.update!(patient_id: real_patient_id) }
+      patient.prescription_drugs.each { |record| record.update!(patient_id: real_patient_id) }
+      patient.medical_history&.update!(patient_id: real_patient_id)
```

Comment:
I assume we don't need to add `Address` or `PatientPhoneNumber` here since we didn't find any updates to this?